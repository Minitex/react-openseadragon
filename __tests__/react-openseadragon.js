import { MemoryRouter } from 'react-router-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import ReactOpenSeadragon from '../src/react-openseadragon-route';

describe('OpenSeadragon', () => {
  it('should render correctly', () => {
    window.OpenSeadragon = jest.fn();
    window.OPENSEADRAGONVIEWER = {
      addHandler: () => { return false }
    }

    const methods = {
      addHandler: () => false,
      goToPage: () => false,
    };

    Object.defineProperty(window, 'OpenSeadragon', {
      value: jest.fn(() => methods),
    });

    const config = {
      type: 'image',
      label: 'Image',
      focus: true,
      include_controls: true,
      sequenceMode: true,
      showReferenceStrip: true,
      defaultZoomLevel: 0,
      tileSources: [
        'https://stacks.stanford.edu/image/iiif/hg676jb4964%2F0380_796-44/info.json',
        'https://ids.lib.harvard.edu/ids/iiif/25286610/info.json',
      ],
      tocs: [
        'MLK',
        'A Statue',
      ],
    };

const component = renderer.create(
  <MemoryRouter initialEntries={['/image/0']} initialIndex={1} >
    <ReactOpenSeadragon config={config} />
  </MemoryRouter>);

    const jsonComponent = component.toJSON();
    expect(jsonComponent).toMatchSnapshot();

  });
});
