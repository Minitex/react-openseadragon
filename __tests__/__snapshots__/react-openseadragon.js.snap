// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenSeadragon should render correctly 1`] = `
<div>
  <div>
    <div>
      <div>
        <div
          className="row image-nav"
        >
          <ul
            className="nav nav-pills"
          >
            <li>
              <div
                className="toc-select"
              >
                <label
                  htmlFor="toc"
                >
                  Table of Contents:
                </label>
                <select
                  name="toc"
                  onChange={[Function]}
                >
                  <option
                    selected="selected"
                    value={0}
                  >
                    1
                    . 
                    MLK
                  </option>
                  <option
                    value={1}
                  >
                    2
                    . 
                    A Statue
                  </option>
                </select>
              </div>
            </li>
          </ul>
        </div>
        <div
          className="osd col-md-12"
        >
          <div
            className="openseadragon"
            id="osd-viewer"
          >
            <ul
              className="osd-toolbar"
            >
              <li>
                <a
                  id="zoom-in"
                >
                  <i
                    className="glyphicon glyphicon-zoom-in"
                  />
                </a>
              </li>
              <li>
                <a
                  id="zoom-out"
                >
                  <i
                    className="glyphicon glyphicon-zoom-out"
                  />
                </a>
              </li>
              <li>
                <a
                  id="reset"
                >
                  <i
                    className="glyphicon glyphicon-home"
                  />
                </a>
              </li>
              <li>
                <a
                  id="full-page"
                >
                  <i
                    className="glyphicon glyphicon-resize-full"
                  />
                </a>
              </li>
              <li>
                <a
                  id="previous"
                >
                  <i
                    className="glyphicon glyphicon-arrow-left"
                  />
                </a>
              </li>
              <li>
                <a
                  id="next"
                >
                  <i
                    className="glyphicon glyphicon-arrow-right"
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
